edb/edgeql/parser/grammar/expressions.py

```diff
@@ -293,9 +293,31 @@ def reduce_Insert(self, *kids):
                 len(subj_path.steps) == 1 and  \
                 isinstance(subj_path.steps[0], qlast.ObjectRef):
             objtype = subj_path.steps[0]
+        elif isinstance(subj_path, qlast.IfElse):
+            # Insert attempted on something that looks like a conditional
+            # expression. Aside from it being an error, it also seems that
+            # the intent was to insert something conditionally, which
+            # cannot be done this way.
+            raise errors.EdgeQLSyntaxError(
+                f"INSERT statements cannot be used inside conditional "
+                f"expressions",
+                hint=(
+                    f"To resolve this try to factor out the mutation "
+                    f"expression into the top-level WITH block. "
+                    f"Alternatively, use a FOR loop to iterate over "
+                    f"some data that's constructed using conditional "
+                    f"expressions and then supplied to the INSERT."
+                ),
```

Comment:
Is the idea that once we support conditional inserts, we'll do something different here?